src/floom/system.js

```diff
@@ -75,6 +75,13 @@ import Integrator from "./integrator.js";
 		this.mapPropertiesToGrid();
 		this.sumUpPerMaterialGradients();
 
+		// https://github.com/hughsk/clamp/blob/master/index.js
+		function clamp(value, min, max) {
```

Comment:
As we only have a single use for clamp, it should be scoped accordingly. Creating a new clamp function in the inner loop can be well optimized by V8's JIT and shouldn't have a negative performance impact.

Reply:
up scope for a single initialization